pkg/stanza/operator/parser/container/parser.go

```diff
@@ -104,48 +97,28 @@ func (p *Parser) Process(ctx context.Context, entry *entry.Entry) (err error) {
 			p.asyncConsumerStarted = true
 		})
 
-		// Short circuit if the "if" condition does not match
-		skip, err := p.Skip(ctx, entry)
-		if err != nil {
-			return p.HandleEntryError(ctx, entry, err)
-		}
-		if skip {
-			return p.Write(ctx, entry)
-		}
-
 		if format == containerdFormat {
 			// parse the message
-			err = p.ParserOperator.ParseWith(ctx, entry, p.parseContainerd)
+			err = p.ParserOperator.ParseWithCallback(ctx, entry, p.parseContainerd, p.cbGoTimeLayout)
 			if err != nil {
-				return fmt.Errorf("failed to parse containerd log: %w", err)
+				return p.HandleEntryError(ctx, entry, stanzaerr.Wrap(err, "containerd log"))
```

Comment:
Isn't the error already handled inside the `ParseWithCallback`? I would assume we should return `nil` here as the entry has already been either moved forward or dropped.